src/test/scala/mesosphere/marathon/integration/setup/ProcessKeeper.scala

```diff
@@ -169,16 +171,25 @@ object ProcessKeeper {
     }
   }
 
+  def stopProcess(name: String): Option[Int] = {
+    log.info(s"Stop Process $name")
+    processes.get(name).map { process =>
+      Try(process.destroy())
+      processes -= name
+      process.exitValue()
```

Comment:
Isn't this affected by the race conditions mentioned in L188?

It looks odd that stopAllProcesses actually doesn't call this method for each process.